src/modules/ce/components/ReferralStepCard.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { useMemo } from 'react';
+import { useNavigate } from 'react-router-dom';
+import ReferralStepCardInner from './ReferralStepCardInner';
+import ButtonLink, { ButtonLinkProps } from '@/components/elements/ButtonLink';
+import LoadingButton from '@/components/elements/LoadingButton';
+import { GoToIcon } from '@/components/elements/SemanticIcons';
+import useSafeParams from '@/hooks/useSafeParams';
+import { useReferralContext } from '@/modules/ce/components/ReferralPage';
+import ReferralStepAssignee from '@/modules/ce/components/ReferralStepAssignee';
+import { cache } from '@/providers/apolloClient';
+import { ProjectDashboardRoutes } from '@/routes/routes';
+import {
+  CeReferralStepStatus,
+  CeReferralStepSummaryFieldsFragment,
+  GetCeReferralStepDocument,
+  useStartCeReferralStepMutation,
+} from '@/types/gqlTypes';
+import { generateSafePath } from '@/utils/pathEncoding';
+
+interface Props {
+  step: CeReferralStepSummaryFieldsFragment;
+}
+
+const ReferralStepCard: React.FC<Props> = ({ step }) => {
+  const { projectId, opportunityId } = useSafeParams() as {
+    projectId: string;
+    opportunityId: string;
+  };
+  const { referral } = useReferralContext();
+  const navigate = useNavigate();
+
+  const { name, status } = step;
+
+  const [startStepMutation, { loading, error }] =
+    useStartCeReferralStepMutation({
+      variables: {
+        referralId: referral.id,
+        stepId: step.stepId || '',
+      },
+      onCompleted: (data) => {
+        if (data.startCeReferralStep?.step) {
+          const step = data.startCeReferralStep?.step;
+
+          // The step returned from the mutation is now auto added to the Apollo cache.
+          // Here we are writing it to the cache specifically for the GetCeReferralStepDocument query,
+          // so that when we navigate to the Step page, we don't do a double-fetch for the data we already have.
+          cache.writeQuery({
```

Comment:
The explicit write is a bit of a code smell to me, but I am guessing it's needed because `CeReferralStep.id` is the cache key but `CeReferralStep.stepId` is the id we're using to fetch.. a little funky but seems ok, we might need to come back to those cache keys in the future if there are issues